src/compiler/transformers/ts.ts

```diff
@@ -1715,27 +1717,35 @@ namespace ts {
                 case SyntaxKind.UnionType:
                     {
                         const unionOrIntersection = <UnionOrIntersectionTypeNode>node;
-                        let serializedUnion: Identifier;
+                        let serializedUnion: Identifier | VoidExpression;

```

Comment:
We should make the serializeType* family strongly typed when it comes to return value. today it returns `Expression`